test/utils.sh

```diff
@@ -158,19 +158,19 @@ parse_test_output() {
 }
 
 # The function will be used by the tekton tasks of build-definitions
-# It need tekton result path as parameter
+# It need tekton result path as parameter when generating TEST_OUTPUT task result is needed
 handle_error()
 {
   exit_code=$?
   if [ "${exit_code}" -ne 0 ]; then
-    # The tekton task result path
-    TEST_OUTPUT_PATH=$1
-    if [ -z "$TEST_OUTPUT_PATH" ]; then
-      echo "Missing parameter TEST_OUTPUT_PATH" >&2
-      exit 2
+    if [ $# == 0 ]; then
+      echo "Unexpected error: Script errored at command: ${BASH_COMMAND}."
+      exit 0
     fi
 
-    note="Unexpected error: Script errored at command: ${BASH_COMMAND}."
+    # The tekton task result path
+    TEST_OUTPUT_PATH=$1
+    note="Task experienced unexpected error in previous step. For details, check Tekton task result and log."
```

Comment:
I don't understand this note.
Unexpected error was in current step

Reply:
Real unexpected error occurs in previous step, this step depends on it so that it can't run successfully.

Reply:
How this function can know what happened in previous step?

Reply:
This function is called in these steps, but only need the `TEST_OUTPUT` parameter for the step it need to generate this result with `Task experienced unexpected error in previous step. For details, check Tekton task result and log.` as `note`. In previous step, it will only print `Unexpected error: Script errored at command: ${BASH_COMMAND}.`

Reply:
```
Step-1
trap handle_error EXIT
echo "this task finished sucesfully"

step-2
trap handle_error $(test_output.path) EXIT
exit 2  # this task failed
```

will return: `{result: "ERROR", "note":"Task experienced unexpected error in previous step. For details, check Tekton task result and log.", ....}`

Reply:
Original note should stay, no changes in note are needed
```
note="Unexpected error: Script errored at command: ${BASH_COMMAND}."
```

Reply:
It makes sense, let me update it. Thanks.

Reply:
Updated.